### ISSUE TITLE
PersonPicture crashes

### ISSUE THREAD

**Issue by TopperDEL** [unoplatform/uno#3591]
## Current behavior

PersonPicture crashes if it cannot generate the initials for an empty name. I get that stack trace:
```
07-21 21:17:40.063 I/mono-stdout(16844): System.NullReferenceException: Object reference not set to an instance of an object.
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.Controls.InitialsGenerator.GetCharacterType (System.String str) [0x0005c] in D:\a\1\s\src\Uno.UI\UI\Xaml\Controls\PersonPicture\InitialsGenerator.cs:261 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.Controls.InitialsGenerator.InitialsFromDisplayName (System.String contactDisplayName) [0x00000] in D:\a\1\s\src\Uno.UI\UI\Xaml\Controls\PersonPicture\InitialsGenerator.cs:111 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.Controls.PersonPicture.OnDisplayNameChanged (Windows.UI.Xaml.DependencyPropertyChangedEventArgs args) [0x00000] in D:\a\1\s\src\Uno.UI\UI\Xaml\Controls\PersonPicture\PersonPicture.cs:610 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.Controls.PersonPicture.OnPropertyChanged (Windows.UI.Xaml.DependencyPropertyChangedEventArgs args) [0x0004d] in D:\a\1\s\src\Uno.UI\UI\Xaml\Controls\PersonPicture\PersonPicture.cs:597 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.Controls.PersonPicture.<.ctor>b__27_0 (Uno.UI.DataBinding.ManagedWeakReference s, Windows.UI.Xaml.DependencyProperty p, Windows.UI.Xaml.DependencyPropertyChangedEventArgs e) [0x00000] in D:\a\1\s\src\Uno.UI\UI\Xaml\Controls\PersonPicture\PersonPicture.cs:92 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.DependencyObjectStore+<>c__DisplayClass141_0.<CreateWeakDelegate>b__0 (Uno.UI.DataBinding.ManagedWeakReference instance, Windows.UI.Xaml.DependencyProperty s, Windows.UI.Xaml.DependencyPropertyChangedEventArgs e) [0x00000] in D:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectStore.cs:1418 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.DependencyObjectStore.InvokeCallbacks (Windows.UI.Xaml.DependencyObject actualInstanceAlias, Windows.UI.Xaml.DependencyProperty property, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails, System.Object previousValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences previousPrecedence, System.Object newValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences newPrecedence, System.Boolean bypassesPropagation) [0x00115] in D:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectStore.cs:1548 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.DependencyObjectStore.RaiseCallbacks (Windows.UI.Xaml.DependencyObject actualInstanceAlias, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails, System.Object previousValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences previousPrecedence, System.Object newValue, Windows.UI.Xaml.DependencyPropertyValuePrecedences newPrecedence) [0x0005b] in D:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectStore.cs:1446 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.DependencyObjectStore.InnerSetValue (Windows.UI.Xaml.DependencyProperty property, System.Object value, Windows.UI.Xaml.DependencyPropertyValuePrecedences precedence, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails) [0x001a5] in D:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectStore.cs:491 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.DependencyObjectStore.SetValue (Windows.UI.Xaml.DependencyProperty property, System.Object value, Windows.UI.Xaml.DependencyPropertyValuePrecedences precedence, Windows.UI.Xaml.DependencyPropertyDetails propertyDetails) [0x0002e] in D:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectStore.cs:424 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.DependencyObjectExtensions.SetValue (System.Object instance, Windows.UI.Xaml.DependencyProperty property, System.Object value, System.Nullable`1[T] precedence) [0x00000] in D:\a\1\s\src\Uno.UI\UI\Xaml\DependencyObjectExtensions.cs:202 
07-21 21:17:40.064 I/mono-stdout(16844):   at NutzMich.Droid.MetadataBuilder_442.SetDisplayName (System.Object instance, System.Object value, System.Nullable`1[T] precedence) [0x00000] in E:\NutzMich\NutzMich\NutzMich.Droid\obj\Debug\90\g\BindableTypeProvidersSourceGenerator\BindableMetadata.g.cs:21570 
07-21 21:17:40.064 I/mono-stdout(16844):   at (wrapper delegate-invoke) <Module>.invoke_void_object_object_Nullable`1<DependencyPropertyValuePrecedences>(object,object,System.Nullable`1<Windows.UI.Xaml.DependencyPropertyValuePrecedences>)
07-21 21:17:40.064 I/mono-stdout(16844):   at Uno.UI.DataBinding.BindingPropertyHelper+<>c__DisplayClass39_9.<InternalGetValueSetter>b__14 (System.Object instance, System.Object value) [0x00000] in D:\a\1\s\src\Uno.UI\DataBinding\BindingPropertyHelper.cs:829 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.Data.BindingExpression.SetTargetValue (System.Object value) [0x0000f] in D:\a\1\s\src\Uno.UI\DataBinding\BindingExpression.cs:409 
07-21 21:17:40.064 I/mono-stdout(16844):   at Windows.UI.Xaml.Data.BindingExpression.SetTargetValueSafe (System.Object v, System.Boolean useTargetNullValue) [0x00102] in D:\a\1\s\src\Uno.UI\DataBinding\BindingExpression.cs:574 
```
In this case sometimes it works and sometimes not. The person-name is async loaded and seems to be there when it works and leads to crashes if it is still loading.

## Expected behavior

PersonPicture can handle a propably empty Person-Name.

## How to reproduce it (as minimally and precisely as possible)

I'm not quite sure, but I guess it crashes already when you just place an empty `<PersonPicture/>` into a page.

## Environment

Nuget Package: v3.0.0-dev.783

Package Version(s):

Affected platform(s):

- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

Visual Studio:

- [ ] 2017 (version: )
- [X] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by MartinZikmund** [comment]
@TopperDEL I have debugged the code and found a bug that caused this `NullRefException`. However, the cause of this was that the `DisplayName` property was `null`, which is not allowed in UWP either. Can you please verify the code crashes on Windows as well? In case not, please provide with some minimal repro, as the empty `<PersonPicture />` does not crash for me.

**Comment by TopperDEL** [comment]
Thanks @MartinZikmund for digging into this issue!
I've only used the PersonPicture together with a binding and it may have been that the bound value for DisplayName was "null". I just tried it again with UWP and bound DisplayName to a "null"-string - that worked. But setting the DisplayName in code to "null" lead to an exception.

So: not sure what to do here. When using Binding there is no exception on UWP but there has been one obviously with Android.
